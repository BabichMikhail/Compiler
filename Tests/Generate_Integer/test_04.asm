extern _printf
section .data
	fmt : times 128 db 0
section .text
global _main
_main:
push 20
push 10
mov eax, '%d'
mov [fmt + 0], eax
mov eax, ' %d'
mov [fmt + 2], eax
mov eax, 0xA
mov [fmt + 5], eax
mov eax, 0x0
mov [fmt + 6], eax
push fmt
call _printf
add esp, 12
push 25
push 15
mov eax, '%d'
mov [fmt + 0], eax
mov eax, ' %d'
mov [fmt + 2], eax
mov eax, 0xA
mov [fmt + 5], eax
mov eax, 0x0
mov [fmt + 6], eax
push fmt
call _printf
add esp, 12
ret