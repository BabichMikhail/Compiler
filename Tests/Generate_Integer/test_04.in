begin
	writeln(10, 20);
	writeln(15, 25);
end.