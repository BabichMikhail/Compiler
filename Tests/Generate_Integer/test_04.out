10 20
15 25
